### PR TITLE
Remove weak linking to fix downstream errors

### LINKED ISSUE #34: Missing link fixing for openblas on OSX clang
I'm having problems building a [downstream package](https://github.com/conda-forge/msprime-feedstock/pull/18) which depends on GSL, and I think it must be something to do with GSL packaging.

On the standard toolchain, we get lots of "Fixing linking of .." messages from conda build [here](https://travis-ci.org/conda-forge/msprime-feedstock/jobs/459904968#L504). This includes ``
libopenblas.dylib`` right after ``libgsl.23.dylib``.

However, on the ``osx_c_compilerclang`` toolchain, we seem to be missing a bunch of these ``Fixing linking of .." message from conda build [here](https://travis-ci.org/conda-forge/msprime-feedstock/jobs/459904966#L545). In particular, the linkage for openblas doesn't get fixed.

Later, the [build fails](https://travis-ci.org/conda-forge/msprime-feedstock/jobs/459904966#L611) 
because of a missing symbol ``_cblas_caxpy`` in GSL. 

Is this a known issue with conda build on OSX? Any pointers on how to track this down would be much appreciated;  unfortunately I don't have access to an OSX machine and know very little about its linking system quirks.

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR fixes #34 by implementing [@isuruf's patch](https://github.com/conda-forge/gsl-feedstock/issues/34#issuecomment-449305702).

I have tested this locally, and it does indeed fix the downstream issues with linking other libraries. I don't really understand the implications of this change, however.

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@isuruf, your patch worked fine on my mac, but is killing the build here. Any chance you can take a look (you clearly understand this much, much better than I do)?

**Comment by duncanmmacleod** [comment]
Actually, I presume this is just because the patch is being applied also on linux, which breaks. I will modify to only set `LIBS` on osx.

**Comment by jeromekelleher** [comment]
I don't have access to a mac unfortunately @duncanmmacleod, so I'll have to wait until the packages wind their way through the pipeline. Fantastic that we're seeing light at the end of the tunnel here!

**Comment by jeromekelleher** [comment]
This might not be the place to bring it up, but I'm still confused about what's happening here:
```
 rm "$PREFIX"/lib/libgslcblas.*
 ln -s "$PREFIX/lib/libopenblas${SHLIB_EXT}" "$PREFIX/lib/libgslcblas${SHLIB_EXT}"
```
Since we're requiring openblas, and we're now linking directly against openblas for OSX, why are we doing this funny workaround of tricking GSL into thinking ``openblas`` is actually ``gslcblas``?

**Comment by duncanmmacleod** [comment]
@conda-forge/gsl, @isuruf, can you give this a review. @jeromekelleher has a separate PR in #38 that goes further, but fails to link properly for me on macos.

File: recipe/build.sh [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
 
+# https://github.com/conda-forge/gsl-feedstock/issues/34#issuecomment-449305702
+if [ "$(uname)" == "Darwin" ]; then
```
Comment (isuruf) [inline recipe/build.sh]: Can you do this for all platforms?

**Pull request merged**